src/NzbDrone.Core/Extras/ExistingExtraFileService.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Extras.ExtraFiles;
+using NzbDrone.Core.Extras.Metadata;
+using NzbDrone.Core.Extras.Subtitles;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.MediaFiles.Events;
+using NzbDrone.Core.Messaging.Events;
+using NzbDrone.Core.Parser;
+using NzbDrone.Core.Tv;
+
+namespace NzbDrone.Core.Extras
+{
+    public class ExistingExtraFileService : IHandle<SeriesScannedEvent>
+    {
+        private readonly IExtraFileService _extraFileService;
+        private readonly IDiskProvider _diskProvider;
+        private readonly IExistingMetadataService _existingMetadataService;
+        private readonly IExistingSubtitleService _existingSubtitleService;
+        private readonly IParsingService _parsingService;
+        private readonly Logger _logger;
+
+        public ExistingExtraFileService(IExtraFileService extraFileService,
+                                        IDiskProvider diskProvider,
+                                        IExistingMetadataService existingMetadataService,
+                                        IExistingSubtitleService existingSubtitleService,
+                                        IParsingService parsingService,
+                                        Logger logger)
+        {
+            _extraFileService = extraFileService;
+            _diskProvider = diskProvider;
+            _existingMetadataService = existingMetadataService;
+            _existingSubtitleService = existingSubtitleService;
+            _parsingService = parsingService;
+            _logger = logger;
+        }
+
+        private List<ExtraFile> ProcessFiles(Series series, List<string> filesOnDisk)
+        {
+            var extraFiles = new List<ExtraFile>();
+
+            foreach (var possibleExtraFile in filesOnDisk)
+            {
+                var localEpisode = _parsingService.GetLocalEpisode(possibleExtraFile, series);
+
+                if (localEpisode == null)
+                {
+                    _logger.Debug("Unable to parse subtitle file: {0}", possibleExtraFile);
```

Comment:
How do you know that this file is a "subtitle" file?  Perhaps the message should read "extra" instead of "subtitle".